.github/workflows/deploy.yml

```diff
@@ -0,0 +1,86 @@
+name: 'Deploy to PaaS'
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'The environment to deploy to eg: qa, staging, production etc'
+        required: true
+      sha:
+        description: Commit sha to be deployed
+        required: true
+
+jobs:
+  deploy:
+
+    name: Deploy to PaaS
+    env:
+        DOCKER_REPO: dfedigital/register-trainee-teacher-data
+        DOCKER_IMAGE: register-trainee-teacher-data
+
+    if: ${{ !startsWith(github.event.inputs.environment, 'dfe-twd-rttd-pr') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: softprops/turnstyle@v1
+        name: Wait for other inprogress deployment runs
+        env:
+          GITHUB_TOKEN: ${{ secrets.API_TOKEN_FOR_GITHUB_ACTION }}
+
+      - name: Start ${{ github.event.inputs.environment }} Deployment
+        uses: bobheadxi/deployments@v0.4.2
+        id: deployment
+        with:
+          step: start
+          token: ${{ secrets.API_TOKEN_FOR_GITHUB_ACTION }}
+          env: ${{ github.event.inputs.environment }}
+          ref: ${{ github.event.inputs.sha }}
+
+
+
+      - name: print env variables
+        run: |
+          GIT_SHA_SHORT=$(echo ${GITHUB_SHA} | cut -c 1-7)
+          echo "SHA_TAG=$GIT_SHA_SHORT" >> $GITHUB_ENV
```

Comment:
can be removed? and just use `${{ github.event.inputs.sha }}` where required?
and I don't think its safe to use `$GITHUB_SHA` as it would refer to the HEAD sha of the branch from which the workflow is run, so you would end up releasing the HEAD commit irrespective of the `${{ github.event.inputs.sha }}` being passed as input.

Reply:
done...